src/GitLabApiClient/IGitLabHttpFacade.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using GitLabApiClient.Models.Uploads.Requests;
+using GitLabApiClient.Models.Uploads.Responses;
+using GitLabApiClient.Models.Users.Responses;
+
+namespace GitLabApiClient.Http
+{
+    internal interface IGitLabHttpFacade
```

Comment:
Thanks, but I don't think we should use interface for GitlabHttpFacade:

1. It's a header interface with lots of methods
2. We don't get much sense of "it will work in production" if we write tests in this "layer". I suggest writing integration tests for GitLab running in docker container, so we know it truly works. 
3. We have almost no logic in places where this interface is used, so nothing to test there

Reply:
I can remove the interface, but I cannot agree with you:
1. There can be many causes for the failure in the integration test: docker problems, Gitlab api changes, coding errors etc
2. By having a unit test at this level, we help others to faster detect the root cause for the failure (faster execution than integration test, complete control of the environment).
I would propose to leave the interface and the original test and extend the test suite with an integration test (docker etc).
What do you think about it?

Reply:
Could you, please, reply to my comment?

Reply:
Sorry for  the delay

Reply:
1. Yes, there can be more issues with integration tests than with mock type tests. We just need to perfect the docker container tests. And I think tests should fail if API changes, that way we don't ship not working code to the clients. Docker container is used for a specific version of GitLab and its completely new every time we start it so this reduces the random fails risk. Currently integration tests are implemented but we need to investigate if they are reliable, they supposed to run on Travis.

2. If we have tests on this level we simply test if we correctly call the IGitLabHttpFacade this is not enough, it does not mean our API client communicates successfully with the API.

if we have code like this:
```public async Task<Commit> GetAsync(string projectId, string sha) =>
           await _httpFacade.Get<Commit>(CommitsBaseUrl(projectId) + "/" + sha);
```
and ```mock _httpFacade.Get<T>``` then our test does not really go trough real code (except some trivial code like CommitsbaseUrl(...) and strings concatenation)

So I suggest to have this kind of tests one level deeper, then they will be useful so mock HttpClient's HttpClientHandler (for ex. https://www.meziantou.net/mocking-an-httpclient-using-an-httpclienthandler.htm, but there should be libraries for that) and they will provide better sense of 'it works in production' if we copy paste the response's correctly from Gitlab docs

So I suggest removing IGitlabHttpFacade and mocking one level deeper, in the level we don't have control.

Reply:
Could you give another look? I have deleted IGitlabHttpFacade + implemented tests by mocking HttpClientHandler with FakeItEasy.

Reply:
Thanks, will do :)

Reply:
Looks good :) we will need to reduce the boilerplate needed for this kind of testing in the future. Merging and pushing new version for now